app/bundles/LeadBundle/Tests/Entity/LeadListRepositoryFunctionalTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\LeadBundle\Tests\Entity;
+
+use Mautic\CoreBundle\Test\AbstractMauticTestCase;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Entity\LeadList;
+use Mautic\LeadBundle\Entity\LeadListRepository;
+use Mautic\LeadBundle\Entity\ListLead;
+
+class LeadListRepositoryFunctionalTest extends AbstractMauticTestCase
+{
+    public function testCheckLeadSegmentsByIds(): void
+    {
+        $lead     = $this->createLead();
+        $segmentA = $this->createSegment();
+        $segmentB = $this->createSegment('B');
+        $this->createSegmentMember($segmentA, $lead);
+        $this->createSegmentMember($segmentB, $lead, true);
+
+        /** @var LeadListRepository $leadListRepository */
+        $leadListRepository = $this->em->getRepository(LeadList::class);
+
+        $result = $leadListRepository->checkLeadSegmentsByIds($lead, [$segmentA->getId()]);
+        $this->assertTrue($result);
+
+        $result = $leadListRepository->checkLeadSegmentsByIds($lead, [$segmentB->getId()]);
+        $this->assertFalse($result);
+
+        $result = $leadListRepository->checkLeadSegmentsByIds($lead, [$segmentA->getId(), $segmentB->getId()]);
```

Comment:
To test also the option that the contact is not a member of a segment. Plus, using assert over type hint.
```suggestion
        $segmentB = $this->createSegment('B');
        $segmentC = $this->createSegment('C');
        $this->createSegmentMember($segmentA, $lead);
        $this->createSegmentMember($segmentB, $lead, true);

        $leadListRepository = $this->em->getRepository(LeadList::class);
        \assert($leadListRepository instanceof LeadListRepository);

        $result = $leadListRepository->checkLeadSegmentsByIds($lead, [$segmentA->getId()]);
        $this->assertTrue($result);

        $result = $leadListRepository->checkLeadSegmentsByIds($lead, [$segmentB->getId()]);
        $this->assertFalse($result);
        
        $result = $leadListRepository->checkLeadSegmentsByIds($lead, [$segmentC->getId()]);
        $this->assertFalse($result);

        $result = $leadListRepository->checkLeadSegmentsByIds($lead, [$segmentA->getId(), $segmentB->getId(), $segmentC->getId()]);
```